### PR TITLE
OSAC-762: Add Organization lifecycle and authorization boundary E2E tests

### PR DESCRIPTION
**Pull request by obochan-rh**
## Summary

- Add 9 integration tests for Organization lifecycle and authorization boundaries
- Add `KeycloakAdminRequest` helper to the test tool for direct Keycloak Admin API access
- Extract `loginAsBreakGlass` helper to reduce duplication across break-glass tests

## Phase 1: Organization Lifecycle (4 specs)

- **CRUD happy flow** — Create, SYNCED, Get, List, verify in Keycloak, Delete, verify removal
- **Break-glass auth and RBAC** — Login as break-glass user, verify private API denied
- **Duplicate org name fails** — AlreadyExists error on same-name creation
- **Rename org is rejected** — InvalidArgument on name change attempt

## Phase 2: Authorization Boundaries (5 specs)

- **Regular user cannot create org** — PermissionDenied on private API
- **Anonymous request cannot create org** — Unauthenticated/PermissionDenied
- **Break-glass from org-A cannot access org-B** — Cross-tenant isolation
- **Break-glass cannot call admin-only APIs** — Create and Delete denied
- **Break-glass JWT contains correct org membership** — Validates `organization` claim and `tenant-idp-manager` role

## Test plan

- [x] All 115 specs pass locally (`ginkgo run -v it`) — 0 failures
- [x] `go vet ./it/...` passes
- [x] No new dependencies added
- [x] Follows existing test patterns (same helpers, assertions, cleanup)
- [x] CI runs full suite (Helm mode) — 115 Passed | 0 Failed

## Summary by CodeRabbit

* **Tests**
  * Added a comprehensive integration test suite covering organization lifecycle, authorization boundaries, input validation, race conditions, and JWT claim assertions.
  * Added a test helper to perform authenticated admin requests against Keycloak to verify creation, presence, renaming, deletion, and cleanup behavior.

### CODE REVIEW

File: it/it_organization_lifecycle_test.go [hunk 1]
```diff
@@ -0,0 +1,633 @@
...
+
+func verifyOrgInKeycloak(ctx context.Context, name string) {
+	code, body, err := tool.KeycloakAdminRequest(ctx, http.MethodGet,
+		fmt.Sprintf("/organizations?search=%s", name), nil)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(code).To(Equal(http.StatusOK))
+	var kcOrgs []map[string]any
+	Expect(json.Unmarshal(body, &kcOrgs)).To(Succeed())
+	Expect(kcOrgs).ToNot(BeEmpty())
+}
```
[the remark this answers is not in the document]
Comment (obochan-rh) [reply it/it_organization_lifecycle_test.go:81]: done

File: it/it_tool.go [hunk 2]
```diff
@@ -1850,6 +1850,80 @@ func (t *Tool) makeKeycloakTokenSource(ctx context.Context, username, password s
...
+		err = fmt.Errorf("failed to create Keycloak admin token source: %w", err)
+		return
+	}
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs: t.caPool,
+			},
+		},
+	}
```
[the remark this answers is not in the document]
Comment (obochan-rh) [reply it/it_tool.go:1887]: done

**Review by jhernand — APPROVED** [review]
